src/burn/engine/plan.cpp

```diff
@@ -2742,6 +2764,84 @@ static HRESULT FinalizeSlipstreamPatchActions(
     return hr;
 }
 
+static HRESULT PlanDependencyActions(
```

Comment:
There are two commits here.  The first one I wrote for my other cache pull request (#40) that hasn't been merged yet.  It was based on changes from your original wix3 pull request, but it incorporates my comments on that pull request.  I already compared it to your updated pull request in my email to wix-devs a couple of nights ago.

The second one's all new, I didn't see any of it in your pull requests.